src/librbd/cache/ObjectCacherObjectDispatch.cc

```diff
@@ -182,35 +182,34 @@ void ObjectCacherObjectDispatch<I>::shut_down(Context* on_finish) {
 
 template <typename I>
 bool ObjectCacherObjectDispatch<I>::read(
-    uint64_t object_no, const io::Extents &extents, IOContext io_context,
-    int op_flags, const ZTracer::Trace &parent_trace,
-    ceph::bufferlist* read_data, io::Extents* extent_map, uint64_t* version,
+    uint64_t object_no, io::ReadExtents* extents, IOContext io_context,
+    int op_flags, const ZTracer::Trace &parent_trace, uint64_t* version,
     int* object_dispatch_flags, io::DispatchResult* dispatch_result,
     Context** on_finish, Context* on_dispatched) {
   // IO chained in reverse order
   auto cct = m_image_ctx->cct;
-  ldout(cct, 20) << "object_no=" << object_no << " " << extents << dendl;
+  ldout(cct, 20) << "object_no=" << object_no << " " << *extents << dendl;
 
-  if (version != nullptr || extents.size() != 1) {
+  if (version != nullptr || extents->size() != 1) {
```

Comment:
Nit: while we are here, let's add support for multiple extents. You can wrap the provided `on_dispatched` callback with a `C_Gather` context so that you can dispatch multiple read extents to the cache.

Reply:
looks like m_object_cacher->readx already supports multiple extents, so I don't think I will use C_Gather